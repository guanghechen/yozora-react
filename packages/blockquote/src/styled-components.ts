import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
