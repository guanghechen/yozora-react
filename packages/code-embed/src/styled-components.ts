import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
