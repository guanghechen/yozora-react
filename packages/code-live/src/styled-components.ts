import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
