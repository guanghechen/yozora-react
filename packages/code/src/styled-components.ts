import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
