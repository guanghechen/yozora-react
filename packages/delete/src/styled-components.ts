import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
