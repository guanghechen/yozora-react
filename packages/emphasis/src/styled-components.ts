import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
