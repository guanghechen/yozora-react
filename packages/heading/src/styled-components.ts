import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
