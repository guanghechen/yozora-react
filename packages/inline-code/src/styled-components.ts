import 'styled-components'
import type { YozoraInlineCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    inlineCode: YozoraInlineCodeTheme
  }
}
