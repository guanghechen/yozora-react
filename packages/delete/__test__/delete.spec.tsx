import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Delete from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <Delete>
        <span>
          <Delete>{ text }</Delete>
        </span>
      </Delete>
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-text'
    const wrapper = render(
      <Delete className={ className }>
        <span>{ text }</span>
      </Delete>
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<Delete>{ value }</Delete>)
      }).toThrow(/Failed prop type: The prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLSpanElement>()
    const wrapper = mount(
      <Delete ref={ ref } data-value="waw">
        1
      </Delete>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Delete style={{ color: 'orange', fontSize: '16px' }}>
        some text1
        <span>some text2</span>
      </Delete>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        delete: {
          color: 'red',
          fontSize: 18,
          fontWeight: undefined,
          // fontStyle: 'oblique',
          textDecoration: 'dashed',
        }
      }
    }

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <Delete>
          some text1
          <span>some text2</span>
        </Delete>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
