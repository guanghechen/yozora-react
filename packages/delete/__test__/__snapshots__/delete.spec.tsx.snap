// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  color: inherit;
  font-size: inherit;
  font-weight: inherit;
  font-style: inherit;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<del
  class="c0"
  style="color:orange;font-size:16px"
>
  some text1
  <span>
    some text2
  </span>
</del>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c0 {
  color: red;
  font-size: 18;
  font-weight: inherit;
  font-style: inherit;
  -webkit-text-decoration: dashed;
  text-decoration: dashed;
}

<Component
  theme={
    Object {
      "yozora": Object {
        "delete": Object {
          "color": "red",
          "fontSize": 18,
          "fontWeight": undefined,
          "textDecoration": "dashed",
        },
      },
    }
  }
>
  <YozoraDelete>
    <styled.del
      theme={
        Object {
          "yozora": Object {
            "delete": Object {
              "color": "inherit",
              "fontSize": "inherit",
              "fontStyle": "inherit",
              "fontWeight": "inherit",
              "textDecoration": "line-through",
            },
          },
        }
      }
    >
      <del
        className="c0"
      >
        some text1
        <span>
          some text2
        </span>
      </del>
    </styled.del>
  </YozoraDelete>
</Component>
`;
